### PR TITLE
exporter/datadogexporter: add support for metrics as APM Stats

### PR DESCRIPTION
**Pull request by gbbr**
This change adds the APM Stats consumer support into the Datadog Exporter.

Required by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16853

### CODE REVIEW

File: exporter/datadogexporter/metrics_exporter.go [hunk 1]
```diff
@@ -208,5 +211,12 @@ func (exp *metricsExporter) PushMetricsData(ctx context.Context, md pmetric.Metr
 		)
 	}
 
+	if len(sp) > 0 {
+		exp.params.Logger.Debug("exporting APM stats payloads", zap.Any("stats_payloads", sp))
+		for _, p := range sp {
+			exp.apmStatsProcessor.ProcessStats(p, "", "")
```
Comment (mx-psi) [inline exporter/datadogexporter/metrics_exporter.go:217]: We can (at least for traces coming from the OTel SDKs) use the [Telemetry SDK semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk) to get the lang and tracer version values here (the second and third arguments).

Do we want to do that? (Not a blocker for this PR if `ProcessStats` works correctly with empty strings there)

File: exporter/datadogexporter/metrics_exporter_test.go [hunk 2]
```diff
@@ -286,6 +287,67 @@ func Test_metricsExporter_PushMetricsData(t *testing.T) {
...
 		},
+		{
+			metrics: createTestMetricsWithStats(),
+			source: source.Source{
+				Kind:       source.HostnameKind,
+				Identifier: "test-host",
+			},
+			histogramMode: HistogramModeDistributions,
+			hostTags:      []string{"key1:value1", "key2:value2"},
+			expectedSeries: map[string]interface{}{
```
Comment (mx-psi) [inline exporter/datadogexporter/metrics_exporter_test.go:298]: nit: feel free to use `any` here if you feel like it :)

**Review by mx-psi — APPROVED** [review]
LGTM, just left two non-blocking comments

**Comment by mx-psi** [comment]
Should we add a changelog here, since this adds functionality visible by users? Presence/absence of this feature may be relevant on setups with multiple Collectors with different versions

File: exporter/datadogexporter/metrics_exporter.go (hunk 1 above)
Comment (gbbr) [reply exporter/datadogexporter/metrics_exporter.go]: This is actually the source of the stats, used for metrics. I added the build info (command + version) here.

File: exporter/datadogexporter/metrics_exporter_test.go (hunk 2 above)
Comment (gbbr) [reply exporter/datadogexporter/metrics_exporter_test.go:298]: Nice! :) I like that. We should probably change that throughout though.

**Comment by gbbr** [comment]
PTAL. The way I see things, this change is useless without the processor. If the processor is added, then that will be in the changelog, and it will signal that there is support for it, if that makes sense.

File: exporter/datadogexporter/metrics_exporter.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/metrics_exporter.go]: Oh, okay! Yes, they were confusing to me at least

**Comment by mx-psi** [comment]
To make the situation I was referring to clearer, let's assume we release this on v0.67.0. Then the following setup will be broken:

```mermaid
graph LR
    subgraph Collector v0.67.0
        A[OTLP<br/> Receiver] --> B[Datadog <br/>Processor] --> C[OTLP<br/> Exporter]
    end

    subgraph Collector v0.60.0
        C --> D[OTLP<br/> Receiver] --> E[Datadog <br/>exporter]
    end
```

and it will be easier for users to know why if we have a changelog that indicates what is the first version of the exporter that had support for this, the same way you added a changelog note on the Datadog Agent.

**Pull request merged**